compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/calls/Candidate.kt

```diff
@@ -101,6 +101,7 @@ class Candidate(
     var usesSuspendConversion: Boolean = false
 
     var argumentMapping: Map<FirExpression, FirValueParameter>? = null
+    var oldToNewArgumentMapping: Map<FirExpression, FirExpression>? = null
```

Comment:
I think it's better to rename that mapping to something more specific, lile `argumentMappingForArraySetCall`, and also make this property nullable

Reply:
I ended up finding a different solution. I removed this and instead we can use `Candidate.argumentMapping` instead of the `Fir*Call.argumentList` to remap vararg arguments to a `FirVarargArgumentExpression`.

The `argumentList` was used to preserve the order, but we can rely on the mapping to have its keys in order. See latest 2 commits.